tpdownes/hpc-toolkit
# Test

- Bump github.com/hashicorp/hcl/v2 from 2.14.0 to 2.14.1
- Pass project_id of subnetwork to batch-job
- Ability to pass non-text vars with cli --vars flag
- Update default MTU for new VPCs
- Handle Unmarshal errors
- Add shared task for recover from terraform failure
- hybrid-slurm documentation for on-premise
- Wait after starup script failure to allow logging to happen
- Automatically pull serial port 1 logs on startup script failure
- Remove blueprint_dir update in int. tests
- Fix Ansible module for upgrading setuptools in HTCondor autoscaler
- Run HTCondor integration test on public example
- Add documentation that cloud-batch-job supports shared vpc
- Update cloud-batch-job doc to make it more clear what is being created
- Add test_apis_enabled to default validator list
- Implement test_apis_enabled validator
- Update minimum Go in Makefile
- Fix mistakes in comments
- Name requirements merging function less generically and improve commenting
- Simplify blueprint requirements merging function
- Ensure that required APIs accept variable project IDs
- Encourage user to enable Service Usage API when Compute Engine API is found to be disabled
- Update README with new information for blueprint validators
- Ensure Toolkit-managed projects are created with bare minimum APIs
- Document required APIs feature in blueprint YAML specification
- Update all tutorials to require Service Usage API
- Add Spack setup runner
- Use user syntax for specifying var.project_id now that expand supports conversion to internal format
- Follow PR review advice to use func handleVariable
- Address misspellings in README
- Relocate validator documentation to dedicated location
- Bump google.golang.org/api from 0.96.0 to 0.98.0
- Document troubleshooting for Slurm startup script timeouts
- Bump github.com/zclconf/go-cty from 1.10.0 to 1.11.0
- Update wait for startup for qsim example to 2400s
- Bump github.com/googleapis/gax-go/v2 from 2.4.0 to 2.5.1
- Ensure test_apis_enabled handles empty API list
- Modify test_apis_enabled unit test now that 0 required APIs case is specially handled
- Fix: passing non-existent cache was causing failures in some projects
- Expose task count on cloud-batch-job module
- Adding mpi_mode to cloud-batch-job module
- Add default Spack log file
- Add slurm v5 integration test for hpc-centos7
- Add nightly test info to the README
- Differentiate login ip parameter in slurm tests
- Rename hybrid slurm doc filenames for clarity
- Support local module required APIs
- Batch modules should require Batch API
- Update centos7 slurmv5 build paramters
- Clarify and compact validator failure message
- Remove extraneous backtick in markdown
- Adding barriers in mpi_mode for cloud-batch-job
- Docs, test & better error msg
- Docs, test & better error msg
- Docs, test & better error msg
- add escape blueprint var capability
- Add escape literal var capability
- Add escape reference into README
- Remove deployment_name var from vpc example as it is passed implicitly
- Add slurm-gcp-v5-ubuntu2004 example to integration tests
- Check multiple espace variables within single string
- Add OpenFOAM v8 recipe to AMD blueprint
- Add OpenFOAM test suite example
- Update to the latest slurm image
- Move v5 ubuntu test to group 4 to balance
- Added a partition with local ssd drives
- fixed typo
- Add variable to turn off public ips on Batch jobs
- Adjust names and comments
- Fail if there is an active filestore during API reset
- Add readme describing cloud-build tools
- Add image support documentation
- Clarify official support, simplify vm-images docs
- List supported images in top level readme
- Push test deployment directory to GCS
- Convert dev bucket name to variable
- Clarify language in image support in main README
- Add test_name to ubuntu slurm v5 test
- Remove Batch startup script workaround
- Add additional network interfaces to vm-instance
- Handle case that cloud-batch-job does not have startup script in metadata
- Improve Batch output instructions and add README to login node
- Add escape blueprint vars func in the writers
- Address feedback
- Make network_interfaces override other network variables
- Fix link to the hpc-cluster-localssd example
- Fix EXAScaler issue with Slurm-on-GCP V5
- Adding fix for pre-existing-network-storage+lustre
- Fix default net interface, formatting improvements
- Add tests
- Refactor mount shell script to take args
- Added Simcenter StarCCM+ tutorial blueprint.
- Formatting updates, add license to blueprint
- Add info on connecting VPC networks to vm-instance
- consolidate functions into hcl_utils
- Add escaped backslash tests
- Move mount command construction inside of shell script
- Fix: logs were not being captured for invalid return status
- Automatically prepend Ansible installation runner if other runners use Ansible
- Remove explicit Ansible installation runners
- Add end-of-file-fixer pre-commit hook to enforce trailing newlines
- Add support for mount options in mount.sh
- Fix EOF in all repo files
- Mount script also makes entry in /etc/fstab
- Update output documentation to avoid explicit installation of Ansible
- Update pre-commit hooks
- Bump github.com/googleapis/gax-go/v2 from 2.5.1 to 2.6.0
- Bump google.golang.org/api from 0.98.0 to 0.99.0
- Bump github.com/spf13/cobra from 1.5.0 to 1.6.0
- Update ansible-lint pre-commit hook
- Remove unnecessary quotation marks from YAML
- Address ansible linting errors in integration tests
- Address ansible linting errors in HTCondor modules
- Add empty fields to network_storage for client install and mount runners
- Add mount and install runners to DDN network_storage
- Add mount and install runners to nfs & filestore network_storage
- Add mount and install runners to pre-existing-network-storage
- Bump github.com/zclconf/go-cty from 1.11.0 to 1.11.1
